magma/bits.py

```diff
@@ -115,236 +78,135 @@ def wire(i, o, debug_info):
             o = bits(o, len(i))
         super().wire(o, debug_info)
 
-    def zext(self, value):
-        from .conversions import zext
-        t = zext(self, value)
-        return t
+    @classmethod
+    def make_constant(self, value, num_bits:tp.Optional[int]=None) -> 'AbstractBitVector':
+        raise NotImplementedError()
 
-    def __getitem__(self, key):
-        from .conversions import bits
-        result = super().__getitem__(key)
-        if isinstance(key, slice):
-            return bits(result)
-        return result
+    def __setitem__(self, index : int, value : AbstractBit):
+        raise NotImplementedError()
 
+    def bvnot(self) -> 'AbstractBitVector':
+        raise NotImplementedError()
```

Comment:
Still need to implement the operators as magma primitives. The plan is to use circuit declarations for them, and "link" in the definitions from a particular mantle implementation.